sys/arduino/serialport.cpp

```diff
@@ -41,19 +42,21 @@ SerialPort::SerialPort(uart_t dev)
 
 int SerialPort::available(void)
 {
-    return (int)rx_buf.avail;
+    return stdio_avail();
 }
 
 void SerialPort::begin(long baudrate)
 {
+    stdio_init();
     /* this clears the contents of the ringbuffer... */
-    ringbuffer_init(&rx_buf, rx_mem, SERIAL_RX_BUFSIZE);
-    uart_init(dev, (uint32_t)baudrate, rx_cb, (void *)&rx_buf);
+    stdio_clear();
+    (void)baudrate; // FIXME Should we use the baudrate, somehow?
```

Comment:
In case of `std_uart` backend, the baudrate is fixed and defined by `STDIO_UART_BAUDRATE`.